client/settings/fallbacks.js

```diff
@@ -54,17 +54,16 @@ export const WC_ASSET_URL = allSettings.wcAssetUrl;
 export const DEFAULT_DATE_RANGE = allSettings.defaultDateRange;
 
 export function getAllSettings() {
-	return {
-		...allSettings,
-		ADMIN_URL,
-		COUNTRIES,
-		CURRENCY,
-		LOCALE,
-		ORDER_STATUSES,
-		SITE_TITLE,
-		WC_ASSET_URL,
-		DEFAULT_DATE_RANGE,
-	};
+	allSettings.ADMIN_URL = ADMIN_URL;
+	allSettings.COUNTRIES = COUNTRIES;
+	allSettings.CURRENCY = CURRENCY;
+	allSettings.LOCALE = LOCALE;
+	allSettings.ORDER_STATUSES = ORDER_STATUSES;
+	allSettings.SITE_TITLE = SITE_TITLE;
+	allSettings.WC_ASSET_URL = WC_ASSET_URL;
+	allSettings.DEFAULT_DATE_RANGE = DEFAULT_DATE_RANGE;
+
+	return allSettings;
```

Comment:
`import * as allSettings from '@woocommerce/wp-admin-settings';` won't get all settings. It will get constants such as `DEFAULT_DATE_RANGE` as well as methods such as `getSetting`. Properties like `max_columns` aren't exported. In order to remove this function, I'd have to change the index file to export the entire `SOURCE` object as well.

Reply:
> import * as allSettings from '@woocommerce/wp-admin-settings'; won't get all settings. It will get constants such as DEFAULT_DATE_RANGE as well as methods such as getSetting.

I think we may have two different definitions of "all settings" here.  For me, the term "all settings" refers to everything the settings api exposes on the `wc.wcSettings`  api which includes a subset of specific properties exported on the `wcSettings` global converted to constants and `getSetting` and `setSetting` for everything else. 

It sounds like your definition of the phrase "all settings" refers to _everything_ exposed on `wcSettings` global from the server which includes any properties added via filter.

Is this correct?  If so, I still stand by what I originally wrote because the main rationale behind `getSetting` was a mechanism for retrieving the setting value for things that aren't exported on the _core_ `@woocommerce/settings` (or `wc.wcSettings`) api but instead are injected via filter. This provides a safe fallback for when/if that filter is removed or not present.  So not only do I not think it necessary to have something like a `getAllSettings` function but I think it is potentially fragile to have it.

> Properties like max_columns aren't exported.

Correct, because they are not intended to be exposed on the _core_ `@woocommerce/settings` api being that they are added via filter.  So to get `max_colums` you would do `getSetting( 'max_columns')`.

## Going Forward

However, with all the above said, I see that you are intentionally wanting to expose _all properties_ exposed on the `wcSettings` global (as opposed to `wc.wcSettings` api) because of the move to make the new `wc/admin/settings` data store the only source for settings after it's initialized.

With that in mind, here's what I suggest be done.

- don't define and export `getAllSettings` from the `fallbacks.js` file.  It is something _specific_ to wc-admin and will not be exposed by core (either via blocks or if/when the `@woocommerce/settings` api is eventually added to woo core).
- define and export `getAllSettings` from `client/settings/index.js` directly.

Reply:
>It sounds like your definition of the phrase "all settings" refers to everything exposed on wcSettings global from the server which includes any properties added via filter.

>Is this correct?

Yes, thanks for clarifying and I understand what you mean by:

> the main rationale behind getSetting was a mechanism for retrieving the setting value for things that aren't exported on the core @woocommerce/settings (or wc.wcSettings) api but instead are injected via filter.

The issue is I need a way to get all the data found in `window.wcSettings` into state when the app initializes, as you mentioned.

In `client/settings/index.js` I can define `getAllSettings` like below, but I'm not entirely sure of the consequences with blocks and its manipulation of settings, especially in regards to versions etc. Is this what you had in mind? 

```js
export function getAllSettings() {
    return window.wcSettings;
}
```

Reply:
Actually, right now the only purpose for the `@woocommerce/wc-admin-settings` alias is to provide backward compat for every place the original `wcSettings` global was being used and to connect with the settings api exposed in `@woocommerce/settings` (which may or may not be present depending on what version of Woo core wc-admin is active with).

Since with this pull and the subsequent refactors after this is merged across the wc-admin codebase to switch away from `@woocommerce/wc-admin-settings` to the new data api, you could just pass in `window.wcSettings` as the settings source to `withSettingsHydration`. 

After the refactors over to the new wp.data api are complete across the client application, you can then remove the alias `@woocommerce/wc-admin-settings` configuration and the related logic in `client/settings/`.